### DISCUSSION TITLE
How to configure log level

### DISCUSSION THREAD

**Discussion by harto** [valhalla/valhalla#4128] [Q&A] [answered]
Hi there,
I noticed that the default logging behaviour is a bit noisy. For example, here's what it looks like when the `/status` API is fetched (I'm using this as a healthcheck URL, so it gets hit rather frequently):
```
0 2023/05/26 03:49:33.955452 GET /status HTTP/1.1
2023/05/26 03:49:33.960335 [INFO] Got Loki Request 0
2023/05/26 03:49:33.980137 [INFO] Got Thor Request 0
2023/05/26 03:49:33.984653 [INFO] Got Odin Request 0
0 2023/05/26 03:49:33.988542 200 356
```

Here's what a successfull routing request looks like
```
1 2023/05/26 03:50:03.651378 GET /route?json=%7B%22locations%22%3A%5B%7B%22lon%22%3A153.02757%2C%22lat%22%3A-27.49791%2C%22type%22%3A%22break%22%7D%2C%7B%22lon%22%3A153.03048140755186%2C%22lat%22%3A-27.498412702437655%2C%22type%22%3A%22break%22%2C%22rank_candidates%22%3Afalse%2C%22radius%22%3A50%7D%5D%2C%22directions_options%22%3A%7B%22narrative%22%3Afalse%7D%2C%22costing%22%3A%22bus%22%2C%22output%22%3A%22json%22%7D HTTP/1.1
2023/05/26 03:50:03.654037 [INFO] Got Loki Request 1
2023/05/26 03:50:04.371101 [INFO] Got Thor Request 1
2023/05/26 03:50:04.375631 [INFO] algorithm::bidirectional_a*
2023/05/26 03:50:04.409335 [INFO] Got Odin Request 1
1 2023/05/26 03:50:04.416843 200 1022
```

And here's an unsuccessful routing request:
```
2 2023/05/26 03:53:02.621453 GET /route?json=%7B%22locations%22%3A%5B%7B%22lon%22%3A144.92645%2C%22lat%22%3A-37.79383%2C%22type%22%3A%22break%22%7D%2C%7B%22lon%22%3A144.92986%2C%22lat%22%3A-37.79369%2C%22type%22%3A%22break%22%2C%22rank_candidates%22%3Afalse%2C%22radius%22%3A50%7D%5D%2C%22directions_options%22%3A%7B%22narrative%22%3Afalse%7D%2C%22costing%22%3A%22bus%22%2C%22output%22%3A%22json%22%7D HTTP/1.1
2023/05/26 03:53:02.623527 [INFO] Got Loki Request 2
2023/05/26 03:53:02.646665 [WARN] 400::No suitable edges near location request_id=2
2 2023/05/26 03:53:02.649714 400 228
```

I think what I want is for the `[INFO]` lines to be hidden, but to keep the HTTP request/response lines and the `[WARN]` messages. Is this something I can configure? Thanks 😄

**Comment by nilsnolde** [comment, marked as answer]
Unfortunately logging level is set at compile time, see https://github.com/valhalla/valhalla/blob/efb4d74af3158ec9633970b249cd95a9ca97b033/CMakeLists.txt#L40. I think it’s „WARNING“ you want. If not it’s „WARN“. On the phone right now, you can also look at the macros in ./valhalla/logging.h.

**Reply by harto** [reply]
Ah, that's a shame. I'm using the pre-built Docker image and would prefer not to recompile from source. Thanks for the pointer, anyway 😄

**Reply by nilsnolde** [reply]
I agree it's a bit annoying, but you'd need to only do it once, i.e. 10 mins docker image build, could be worse:)

**Reply by kevinkreiser** [reply]
just for completeness' sake, i'll say that compile time log level is by design. the reason for this is that if it weren't logging would have to be checked at runtime, which means tons of `if`s (branch instructions) in the code. if we can avoid it, we always want to avoid having branch instructions as they are one of the more costly things to do with your CPU (may stall the execution pipeline). given the amount of logging we have in `TRACE` mode, we'd have a ton of conditionally executed code and so its best that, we decide up front at compile time, whether or not we are ok with the penalty.

**Comment by nilsnolde** [comment]
And http logs should still work, they’re from the server part.
